Automattic/wp-calypso
# Checkout: Move existing card payment method to wpcom-checkout package

#### Changes proposed in this Pull Request

This PR moves the existing credit card payment method from the `@automattic/composite-checkout` package into the `@automattic/wpcom-checkout` package and converts it to TypeScript. This is similar to how Bancontact was moved in https://github.com/Automattic/wp-calypso/pull/51690, Giropay in https://github.com/Automattic/wp-calypso/pull/53346, and p24 in https://github.com/Automattic/wp-calypso/pull/53510 This is related to the refactoring of https://github.com/Automattic/wp-calypso/issues/52215

#### Background

The `composite-checkout` package is intended to be a fairly generic checkout-building toolkit. As part of this we originally envisioned it containing various payment methods, but that quickly became a problem as it became clear that the payment methods we use on WordPress.com are pretty specific to our platform. To reduce leaky abstractions, we moved many of our payment methods over to calypso, but some have remained because the cost of moving them seemed too high for the benefit.

As we prepare to publish version 1.0.0 of the `@automattic/composite-checkout` package on npm, I think it's a good time to clean up extraneous parts of the package. The payment methods are the only part of the package that are not fully TypeScript and are also still pretty specific to our WordPress.com infrastructure. Eventually it would be nice to convert them all to TypeScript and put them in a package like `@automattic/wpcom-checkout` (which is intended to be WordPress.com specific). This PR converts one such payment method and moves it there.

#### Testing instructions

- Make sure you have a stored card on your account.
- Make a new purchase in checkout and select a stored card as the payment method.
- Verify that the purchase completes successfully.
